src/main/java/seedu/address/model/Duration.java

```diff
@@ -0,0 +1,85 @@
+package seedu.address.model;
+
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Objects;
+
+/**
+ * A Duration contains a start time and an end time.
+ */
+public class Duration implements Comparable<Duration> {
+
+    private static final DateTimeFormatter TIME_FORMAT = DateTimeFormatter.ofPattern("HHmm");
+    private LocalTime start;
+    private LocalTime end;
+
+    private Duration(LocalTime firstTime, LocalTime secondTime) {
+        boolean isFirstTimeEarly = firstTime.compareTo(secondTime) < 0;
+        this.start = isFirstTimeEarly ? firstTime : secondTime;
+        this.end = isFirstTimeEarly ? secondTime : firstTime;
+    }
+
+    /**
+     * Builds duration from two text representations of time. The method decides the start and end time of the duration
+     * based on the input.
+     * <p>
+     * This methods appends zeros to the front of the input, if
+     * the input is less than 4 digits. For example, "900" will be changed into "0900".
+     *
+     * @param firstTime one of the start or end time
+     * @param secondTime the other start or end time
+     * @return the duration with the specified start and end time
+     */
+    public static Duration parse(String firstTime, String secondTime) {
+        // append front with zero
+        String first = String.format("%04d", Integer.parseInt(firstTime));
+        String second = String.format("%04d", Integer.parseInt(secondTime));
+
+        return new Duration(LocalTime.parse(first, TIME_FORMAT), LocalTime.parse(second, TIME_FORMAT));
```

Comment:
Parse wrong time format may result to error. need to catch DateTimeParseException

Reply:
Good point. Will just throw the exception out, since it's the caller's problem of passing in the wrongly formatted time.